README.md

```diff
@@ -85,6 +85,44 @@ ergonomic to write servers without dealing with sockets or serialization
 directly. Simply implement one of the generated traits, and you're off to the
 races! See the tarpc_examples package for more examples.
 
+## Example: Futures
+
+Here's the same server, implemented using `FutureService`.
+
+```rust
+#![feature(conservative_impl_trait, plugin)]
+#![plugin(tarpc_plugins)]
+
+extern crate futures;
+#[macro_use]
+extern crate tarpc;
+
+use tarpc::util::{FirstSocketAddr, Never};
+use tarpc::sync::Connect;
+
+service! {
+    rpc hello(name: String) -> String;
+}
+
+#[derive(Clone)]
+struct HelloServer;
+
+impl FutureService for HelloServer {
+    type HelloFut = futures::Finished<String, Never>;
```

Comment:
It's needed because the service impler (you) chooses the return type. The trait doesn't know what type of future the service will return. We could avoid the associated types by requiring the return type to be boxed, I.e. `-> Box<Future<Item=String, Error=Never>` but then we'd be forcing the allocation on all users.